### ISSUE TITLE
Edit a saved models parameters

### ISSUE THREAD

**Issue by belfie13** [ollama/ollama#8109]
Currently you would have to create a new model to change any parameters and it's not possible to edit the template at all.

I'm not sure what use the `ollama cp` function has as you can't edit during or after copying so you end up with two identical models.

Would be great to `create` a model, then be able to edit the models `parameters`, `template`, `system` etc... without having to load it and save

you can set the `system` and `parameters` while it's loaded, why not have the ability to edit the models in `ollama list`

**Comment by rick-github** [comment]
```sh
$ ollama cp model model-orig
$ ollama rm model
$ ollama show --modelfile model-orig > Modelfile
$ vi Modelfile
$ ollama create model
```

**Comment by pdevine** [comment]
Hey @belfie13 thanks for the issue. As @rick-github mentioned, you can essentially do everything that you're asking for, you just have to use the `ollama create` command. I'll go ahead and close the issue, but feel free to keep commenting if you feel like I'm missing something.

**Comment by belfie13** [comment]
so thats cool but the only way to edit a model file for an existing model is to create a new model from the existing one and then remove it? is there no way of changing any parameters on model that has been created? im just wanting to reduce the [**copy modelfile** / **edit parameters** / **create model** / **remove old model**] workflow to just [**edit parameters**]

**Comment by rick-github** [comment]
Wrap it in a shell script.

**Comment by belfie13** [comment]
yeah but isn't anyone concerned about writing and erasing gigabytes of data to disk pointlessly? not to mention the 20 seconds of precious time it takes to copy and delete.

**Comment by belfie13** [comment]
@rick-github you wouldn't be able to tell me how to exit vi would you :D ???

**Comment by rick-github** [comment]
The only thing being copied and deleted is the manifest file, a few hundred bytes that points to the blobs that compose the model (weights, parameters, etc).  They are quick and cheap operations.  When you edit the Modelfile, pay attention to the instructions at the top, it will re-use the existing blob and the create will also be quick.

**Comment by belfie13** [comment]
oh damn i did not realize thank you so much for your patience!
